--- users_ldap.py	2014-01-16 11:17:23.000000000 -0500
+++ users_ldap.py.new	2014-01-16 17:39:03.000000000 -0500
@@ -17,6 +17,9 @@
 #    along with this program.  If not, see <http://www.gnu.org/licenses/>.     
 #
 ##############################################################################
+#
+# Modified by Nimesh Jethwa <njethwa@cirruscomputing.com>
+#
 
 import ldap
 import logging
@@ -69,7 +72,9 @@
         :return: an LDAP object
         """
 
-        uri = 'ldap://%s:%d' % (conf['ldap_server'],
+	#Let the superuser specify if ldap or ldaps - Nimesh Jethwa 01/16/2014
+	#uri = 'ldap://%s:%d' % (conf['ldap_server'],
+        uri = '%s:%d' % (conf['ldap_server'],
                                 conf['ldap_server_port'])
 
         connection = ldap.initialize(uri)
@@ -204,7 +209,9 @@
             ondelete='cascade'),
         'ldap_server': fields.char('LDAP Server address', size=64, required=True),
         'ldap_server_port': fields.integer('LDAP Server port', required=True),
-        'ldap_binddn': fields.char('LDAP binddn', size=64,
+	#Change size from 64 to 128 - Nimesh Jethwa 01/16/2014
+	#'ldap_binddn': fields.char('LDAP binddn', size=64,
+        'ldap_binddn': fields.char('LDAP binddn', size=128,
             help=("The user account on the LDAP server that is used to query "
                   "the directory. Leave empty to connect anonymously.")),
         'ldap_password': fields.char('LDAP password', size=64,
